bluepysnap/nodes/node_population.py

```diff
@@ -573,3 +573,18 @@ def h5_filepath(self):
             if self.name in node_conf["populations"]:
                 return node_conf["nodes_file"]
         raise BluepySnapError(f"h5_filepath not found for population '{self.name}'")
+
+    @cached_property
+    def spatial_index(self):
+        """Access to edges spatial index."""
+        try:
+            from spatial_index import open_index
+        except ImportError as e:
+            raise BluepySnapError(
+                (
+                    "Spatial index is for now only available internally to BBP. ",
+                    "It requires `spatial_index`, an internal package.",
+                )
+            ) from e
+
+        return open_index(self.population_config["spatial_segment_index_dir"])
```

Comment:
I'm getting an error that `self.population_config` doesn't exist, does something need to be merged in?

Reply:
that is waiting for https://github.com/BlueBrain/snap/pull/179

Reply:
would it be possible to rebase on that change, and then I'll test again.

Reply:
done

Reply:
cool, looks good; I'll need to add proper support to `libsonata` to handle the path normalization for the spatial index paths, but we should hopefully be able to push this to the finish line shortly.

Reply:
For discussion: since the keys in the configuration proposal at https://github.com/BlueBrain/sonata-extension/pull/17 has been renamed to `spatial_segment_index_dir` and `spatial_synapse_index_dir`, I think it would be clearer to use similar names here, for example `spatial_segment_index` and `spatial_synapse_index`.

Reply:
makes sense :+1: